### PR TITLE
Improve JavaDoc and simplify equals()

### PR DESCRIPTION
**Pull request by minborg**

### CODE REVIEW

File: src/main/java/org/mapdb/Serializer.java [hunk 1]
```diff
@@ -25,108 +24,124 @@
 import java.util.*;
 
 /**
- * Provides serialization and deserialization
+ * This interface specifies how Java Objects are serialized and de-serialized
+ * and also how objects are compared, hashed and tested for equality for use
+ * with MapDB.
+ * <p>
```
Comment (shabanovd) [inline src/main/java/org/mapdb/Serializer.java]: any reason for that `<p>`?

Comment (minborg) [inline src/main/java/org/mapdb/Serializer.java]: It is a way of separating comments so it looks nicer in the IDEs. The paragraph that follows is not related to the overall description of the interface and so, it is separated.

Comment (shabanovd) [inline src/main/java/org/mapdb/Serializer.java]: wander, do it produce invalid html or it fixed by generator?

Comment (jankotek) [inline src/main/java/org/mapdb/Serializer.java]: MapDB uses Dokka generator. I think we 
can use Markdown for javadoc.

Comment (minborg) [inline src/main/java/org/mapdb/Serializer.java]: It is standard now to use the single <p> tag when starting a new paragraph. Check out java.util.Stream for example.

Comment (shabanovd) [inline src/main/java/org/mapdb/Serializer.java]: I understand that and also that html5 do not follow xml standards for some very strange reasons
